lib/src/test/java/vmath/Mat4Test.java

```diff
@@ -194,4 +194,89 @@ void givenInvalidIndices_whenGetIsCalled_thenThrowsException() {
         assertThrows(IndexOutOfBoundsException.class, () -> m.get(0, -1));
         assertThrows(IndexOutOfBoundsException.class, () -> m.get(0, 4));
     }
-}
+
+    @Test
+    void givenIdentityMatrix_whenMultipliedWithAnyMatrix_thenResultIsTheSame() {
+        Mat4 identity = new Mat4(); 
+        Mat4 someMatrix = new Mat4(
+            1.0f, 2.0f, 3.0f, 4.0f,
+            5.0f, 6.0f, 7.0f, 8.0f,
+            9.0f, 10.0f, 11.0f, 12.0f,
+            13.0f, 14.0f, 15.0f, 16.0f
+        );
+
+        Mat4 result = identity.mul(someMatrix);
+        
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                assertEquals(result.get(i, j), someMatrix.get(i, j));
+            }
+        }
+    }
+
+    @Test
+    void givenTwoSpecificMatrices_whenMultiplied_thenResultIsAsExpected() {
+        Mat4 matrixA = new Mat4(
+            1.0f, 2.0f, 3.0f, 4.0f,
+            5.0f, 6.0f, 7.0f, 8.0f,
+            9.0f, 10.0f, 11.0f, 12.0f,
+            13.0f, 14.0f, 15.0f, 16.0f
+        );
+
+        Mat4 matrixB = new Mat4(
+            2.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, 2.0f, 0.0f, 0.0f,
+            0.0f, 0.0f, 2.0f, 0.0f,
+            0.0f, 0.0f, 0.0f, 2.0f
+        );
+
+        Mat4 expected = new Mat4(
+            2.0f, 4.0f, 6.0f, 8.0f,
+            10.0f, 12.0f, 14.0f, 16.0f,
+            18.0f, 20.0f, 22.0f, 24.0f,
+            26.0f, 28.0f, 30.0f, 32.0f
+        );
+
+        Mat4 result = matrixA.mul(matrixB);
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 4; j++) {
+                assertEquals(expected.get(i, j), result.get(i, j));
+            }
+        }
```

Comment:
In your tests please write a separate assertEquals for each index of the matrix (without loops) as in previous pull requests such as [this](https://github.com/toksaitov/vmath/pull/49/files). This is the test standard to ensure there are no bugs within the tests themselves.